gossip/src/cluster_info.rs

```diff
@@ -1045,6 +1045,31 @@ impl ClusterInfo {
         }
     }
 
+    fn find_vote_index_to_evict(&self, should_evict_vote: impl Fn(&Vote) -> bool) -> u8 {
+        let self_pubkey = self.id();
+        let mut num_crds_votes = 0;
+        let vote_index = {
+            let gossip_crds =
+                self.time_gossip_read_lock("gossip_read_push_vote", &self.stats.push_vote_read);
+            (0..MAX_LOCKOUT_HISTORY as u8)
+                .filter_map(|ix| {
+                    let vote = CrdsValueLabel::Vote(ix, self_pubkey);
+                    let vote: &CrdsData = gossip_crds.get(&vote)?;
+                    num_crds_votes += 1;
+                    match &vote {
+                        CrdsData::Vote(_, vote) if should_evict_vote(vote) => {
+                            Some((vote.wallclock, ix))
+                        }
+                        CrdsData::Vote(_, _) => None,
+                        _ => panic!("this should not happen!"),
+                    }
+                })
+                .min() // Boot the oldest evicted vote by wallclock.
+                .map(|(_ /*wallclock*/, ix)| ix)
+        };
+        vote_index.unwrap_or(num_crds_votes)
```

Comment:
hmm should this be:
```if num_crds_votes < MAX_LOCKOUT_HISTORY { num_crds_votes } else { vote_index}```.
that way if the crds table isn't full we prioritize not evicting a vote?

Reply:
I thought about this, but wanted to keep the logic as similar as possible to  the existing logic from `push_vote` where we evict a vote if it doesn't exist in tower, even if there's room in the crds table. Maybe thats slightly more efficient in gosisp than keeping around an unnecessary vote